File: src/main/java/ru/grabovsky/recordkeeping/config/MailConfig.java
package ru.grabovsky.recordkeeping.config;

import freemarker.cache.ClassTemplateLoader;
import freemarker.cache.TemplateLoader;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.view.freemarker.FreeMarkerConfigurer;

/**
 * Configuration required bean for email service
 *
 * @author GrabovskyAlexey
 */
@Configuration
public class MailConfig {
    /**
     * Path to directory with mail message templates
     */
    @Value("${spring.mail.templates.path}")
    private String mailTemplatesPath;

    @Bean
    public FreeMarkerConfigurer freemarkerClassLoaderConfig() {
        freemarker.template.Configuration configuration = new freemarker.template.Configuration(freemarker.template.Configuration.VERSION_2_3_27);
        TemplateLoader templateLoader = new ClassTemplateLoader(this.getClass(), mailTemplatesPath);
        configuration.setTemplateLoader(templateLoader);
        FreeMarkerConfigurer freeMarkerConfigurer = new FreeMarkerConfigurer();
        freeMarkerConfigurer.setConfiguration(configuration);
        return freeMarkerConfigurer;
    }
}

File: notification/src/main/kotlin/ru/grabovsky/recordkeeping/notification/config/MailConfig.kt
package ru.grabovsky.recordkeeping.notification.config

import freemarker.cache.ClassTemplateLoader
import freemarker.cache.TemplateLoader
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.view.freemarker.FreeMarkerConfigurer

/**
 * Configuration required bean for email service
 *
 * @author GrabovskyAlexey
 */
@Configuration
class MailConfig(
    @Value("\${spring.mail.templates.path}")
    private val mailTemplatesPath: String
) {
    @Bean
    fun freemarkerClassLoaderConfig(): FreeMarkerConfigurer {
        val configuration = freemarker.template.Configuration(freemarker.template.Configuration.VERSION_2_3_27)
        val templateLoader: TemplateLoader = ClassTemplateLoader(this.javaClass, mailTemplatesPath)
        configuration.templateLoader = templateLoader
        val freeMarkerConfigurer = FreeMarkerConfigurer()
        freeMarkerConfigurer.configuration = configuration
        return freeMarkerConfigurer
    }
}
